src/index.js

```diff
@@ -67,51 +65,56 @@ const _create = (privateKey, value, seq, isoValidity, validityType, callback) =>
     }
 
     log(`ipns entry for ${value} created`)
-    return callback(null, entry)
-  })
+    return entry
+  } catch (error) {
+    log.error('record signature creation failed')
+    throw errCode('record signature verification failed', ERRORS.ERR_SIGNATURE_CREATION)
```

Comment:
what do you think about handling this error throw in the `sign` function?

Reply:
Yes I think that's better, I refactored it